projects/laji-ui/src/lib/datatable/datatable.component.ts

```diff
@@ -0,0 +1,311 @@
+import { DOCUMENT } from '@angular/common';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, EventEmitter, Inject, Input,
+  OnChanges, OnInit, Output, QueryList, Renderer2, RendererStyleFlags2, SimpleChanges, TemplateRef, ViewChild, ViewChildren } from '@angular/core';
+
+type Keyable = string | number | symbol;
```

Comment:
Could consider splitting the code into multiple files at least in the future since it's quite big already and probably going to get bigger if this datatable is going to be used in all places. For example the models could go to their own file and sorting functions to their own and the pagination to it's own component.

Reply:
I've extracted the paginator into its own component. However the rest of the stuff I have to disagree with, imo splitting files is unnecessary when it doesn't come naturally. Eg. the type defs are directly describing the component so there's no reason they should be namespaced somewhere else.

Btw, I'm hoping that the datatable won't get any (or at least much) bigger.

Reply:
Yeah it's a matter of preference. The reason why I like having the types elsewhere is that it's easier to see the example usage comment or the inputs when they are at the top of the file and that's what I'm usually interested about